### PR TITLE
Adds an "Editor features" section to the editor manual

### LINKED ISSUE #1163: Certain editor settings are relatively undocumented
Went to set up an external editor for GDScript files via "Editor Settings->Text Editor->External" and had to look into the engine source to find the parameters for the "Exec Flags" field.

In the current documentation there isn't a subsection in the editor manual that this information would go, so perhaps making a new file that contains explanations of the some of the more niche or Godot-specific features.

I'm planning on making a PR for this, opening this for discussion on how best to insert this information.

EDIT: I didn't see or didn't look deeply enough into #395 and #965, so this issue is potentially a duplicate.

### PR DESCRIPTION
**Pull request by Dar13**
For now the only feature enumerated in the section is some explanation
of the "external editor" dialog, particularly the "Exec Flags" field.

Could resolve #1163, unless the scope of that issue expands into other areas. I'd be happy to add to this accordingly.

Of course, after I submit this I see #395 and #965. I suppose I should wait until #965 is merged and then expand the "Exec Flags" table field?

### CODE REVIEW

**Comment by pwab** [comment]
Thanks to your PR my [last comment](https://github.com/godotengine/godot-docs/issues/395#issuecomment-364750829) in #395 might get some attention now. 😄 

> I suppose I should wait until #965 is merged and then expand the "Exec Flags" table field?

I wouldn't do it this way. My proposal is to generate the Editor- and Project-Settings pages **automatically**. The information included there (settings name, path, possible values, description) should be integrated into the Godot Engine itself (e.g. for tooltips) and could be easily extracted with a script (see my gist).

The PR #965 and #962 are **manually** created (based on a first proposal by myself) and I vote to not do this and instead use the explained approach of generating these sites.

But I also like the explanations in your PR so we could add an additional page containing this stuff (as you already did). The Editor- and Project-Settings pages could be used as a reference and this extra page explains stuff like _"How to use an external editor"_ or _"How to change gravity in my game on the fly"_.

**TLDR**
I would vote for the following points:

- Merge this PR 👍 
- Comment on [my proposal](https://github.com/godotengine/godot-docs/issues/395#issuecomment-364750829)
- If accepted: I will create a PR including the generated files (and a seperate one with the script to generate it) and therefore close #965 and #962

**Comment by Omicron666** [comment]
Also, I think default value of external editor should be `{file}` and not empty.

**Comment by mhilbrunner** [comment]
Agreeing with @pwab's proposal, I just merged https://github.com/godotengine/godot-docs/pull/1185 which documents the same thing. Once we got the settings docs automated (https://github.com/godotengine/godot-docs/issues/395#issuecomment-364750829) we need to re-visit this.
